### PR TITLE
[Bug Fix] Clear Ramp when Clearing hate

### PR DESCRIPTION
**Pull request by fryguy503**
When clearing a client from the hatelist, also clear them from ramp

### CODE REVIEW

File: zone/mob_ai.cpp [hunk 1]
```diff
@@ -2062,6 +2062,33 @@ void Mob::ClearRampage()
 	RampageArray.clear();
 }
 
+// if force is false, it will not remove feigned clients
+void Mob::RemoveFromRampageList(Mob* mob, bool force)
```
Comment (Kinglykrab) [inline zone/mob_ai.cpp:2066]: Should we name this `remove_feigned` if that's all it's used for?

**Review by Kinglykrab — APPROVED** [review]

File: zone/perl_npc.cpp [hunk 2]
```diff
@@ -204,6 +204,7 @@ bool Perl_NPC_IsOnHatelist(NPC* self, Mob* mob) // @categories Hate and Aggro
 void Perl_NPC_RemoveFromHateList(NPC* self, Mob* mob) // @categories Hate and Aggro
 {
 	self->RemoveFromHateList(mob);
+	self->RemoveFromRampageList(mob);
```
Comment (Kinglykrab) [inline zone/perl_npc.cpp:207]: I see you did Perl but not Lua?

File: zone/entity.cpp [hunk 3]
```diff
@@ -3618,6 +3631,11 @@ void EntityList::ClearFeignAggro(Mob *targ)
 			}
 
 			it->second->RemoveFromHateList(targ);
+
+			if (it->second->GetSpecialAbility(SPECATK_RAMPAGE)) {
+						it->second->RemoveFromRampageList(targ, true);
```
Comment (Kinglykrab) [inline zone/entity.cpp:3636]: Looks like two tabs here?

File: zone/perl_npc.cpp (hunk 2 above)
Comment (fryguy503) [reply zone/perl_npc.cpp:207]: There is not a lua equivalent.  Lua has `RemoveFromHateLists` which has that built in perl has the singular `RemoveFromHateList`

**Pull request merged**